src/lib.rs

```diff
@@ -61,5 +175,20 @@ pub fn downsample(src: &Image, target_width: u32, target_height: u32) -> Vec<u8>
         )
     }
 
+    if let AlphaCoverageSetting::RetainAlphaCoverage { alpha_cutoff } =
+        target_desired_alpha_coverage
+    {
+        assert!(
+            matches!(src.format, Format::RGBA8),
+            "Cannot retain alpha coverage on image with no alpha channel"
+        );
```

Comment:
I think we should just not enter this if-statement if the format does not have an alpha channel. Panicing if we have specified alpha coverage when the format has no alpha channel seems unnecessary, as that is something that can be easily recovered from and is not necessarily invalid.

Reply:
Hm some part of me does not like the fact that we just silently skip it, but it is indeed recoverable. It's just questionable what you're asking if you specify that you want to retain alpha coverage when something like that isn't possible for the given format.